semgrep-core/src/parsing/pfff/java_to_generic.ml

```diff
@@ -421,6 +421,19 @@ and arguments v : G.argument list G.bracket = bracket (list argument) v
 
 and fix_op v = H.conv_incr v
 
+and resource (v : resource) : G.stmt =
+  match v with
+  | Left v ->
+      let ent, v = var_with_init v in
+      G.DefStmt (ent, G.VarDef v) |> G.s
+  | Right e -> G.ExprStmt (expr e, unsafe_fake "") |> G.s
```

Comment:
Can easily avoid `unsafe_fake` here by passing a token to `resource`, which we have readily available in `resources`.